Maskbook/Supporting Files/AppDelegate.swift

```diff
@@ -209,6 +212,41 @@ extension AppDelegate {
             }
             .store(in: &disposeBag)
     }
+    
+    // If you connect to the wallet from imToken, metaMask separately, it may result in duplicate tokens, so let's clean up all the tokens first.
+    private func cleanTokens() {
+        guard !userDefaultSettings.hasCleanTokensForBugfix else {
```

Comment:
Let's say, the user first import a normal wallet and get tokens, later again import a wallet which is from imToken or MetaMask, will this solution also fix the issue?
Also, we should move this `cleanTokens` function into a repository object.

Reply:
Okay, I'll create a repo in provider directory.

Reply:
We can put the operations of CRUD into the `Repository` and consider putting them into the `Service` if they are more complex. `Service` can implement business logic using the `Repository` methods.